### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by dhruvJam**
add login logout [ authentication ] for both user and Restaurant Owner

### CODE REVIEW

**Comment by dhruvJam** [comment]
https://www.awesomescreenshot.com/video/28836869?key=6764014aa3aa772138888e45a0d2f48c

File: src/Components/Header.tsx
Comment (jemis-lakhani) [inline src/Components/Header.tsx:1]: - Code Formatting
- Remove unused imports

File: src/Components/ManageTablesForm.tsx
Comment (jemis-lakhani) [inline src/Components/ManageTablesForm.tsx:1]: - Remove console logs

File: src/app/api/users/login/route.ts
Comment (jemis-lakhani) [inline src/app/api/users/login/route.ts:1]: - Remove console logs.
- DB connection must be async/await.
- Code formatting.

File: src/app/api/users/logout/route.ts
Comment (jemis-lakhani) [inline src/app/api/users/logout/route.ts:1]: - Code formatting

File: src/app/api/users/me/route.ts
Comment (jemis-lakhani) [inline src/app/api/users/me/route.ts:1]: - DB connection must be async/await.
- Code formatting.

File: src/app/api/users/restaurants/route.ts
Comment (jemis-lakhani) [inline src/app/api/users/restaurants/route.ts:1]: - Remove console logs.
- DB connection must be async/await.
- Code formatting.

File: src/app/api/users/signup/route.ts
Comment (jemis-lakhani) [inline src/app/api/users/signup/route.ts:1]: - Remove console logs.
- DB connection must be async/await.
- Code formatting.

File: src/app/api/users/tables/route.ts
Comment (jemis-lakhani) [inline src/app/api/users/tables/route.ts:1]: - DB connection must be async/await.
- Code formatting.

File: src/app/dashboard/page.tsx
Comment (jemis-lakhani) [inline src/app/dashboard/page.tsx:1]: - Can be manage better UI.

File: src/app/layout.tsx
Comment (jemis-lakhani) [inline src/app/layout.tsx:1]: - Code formatting.

File: src/app/login/page.tsx
Comment (jemis-lakhani) [inline src/app/login/page.tsx:1]: - On submit function can be optimize.
- Remove logs.

File: src/app/page.tsx
Comment (jemis-lakhani) [inline src/app/page.tsx:1]: - Code formatting.

File: src/app/profile/[id]/page.tsx
Comment (jemis-lakhani) [inline src/app/profile/[id]/page.tsx:1]: - Code formatting.

File: src/app/signup/page.tsx
Comment (jemis-lakhani) [inline src/app/signup/page.tsx:1]: - Code formatting.
- Remove logs.

File: src/helpers/getDataFromToken.ts
Comment (jemis-lakhani) [inline src/helpers/getDataFromToken.ts:1]: - Code formatting.

File: src/models/tableModel.ts
Comment (jemis-lakhani) [inline src/models/tableModel.ts:1]: - Code formatting.

File: src/models/userModel.ts
Comment (jemis-lakhani) [inline src/models/userModel.ts:1]: - Code formatting.

File: src/app/profile/page.tsx
Comment (jemis-lakhani) [inline src/app/profile/page.tsx:1]: - Code formatting.

File: src/Components/AddRestaurantForm.tsx
Comment (jemis-lakhani) [inline src/Components/AddRestaurantForm.tsx:1]: - Code formatting.
- Remove logs.

File: src/app/login/page.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      toast.success("Login Success");
+      localStorage.setItem("role", response.data.role);
+
+      if (response.data.role === "admin") {
+        router.push("/");
+      } else if (response.data.role === "restaurantOwner") {
+        router.push("/");
+      } else {
+        router.push("/");
+      }
```
Comment (jemis-lakhani) [inline src/app/login/page.tsx]: - Why if/else condition for same output?

**Review by jemis-lakhani — COMMENTED** [review]
- Folder name should start from small letter(camel case).
- Images are must be inside public folder.

File: src/app/api/users/login/route.ts
Comment (dhruvJam) [reply src/app/api/users/login/route.ts:1]: The database connection initialization (dbConfig()) should generally be done outside of your request handler function to ensure it is only called once when the server starts, not on every request.

File: src/app/login/page.tsx (hunk 1 above)
Comment (dhruvJam) [reply src/app/login/page.tsx]: Remove if/else now , Earlier I move users to different routes based on role.

**Comment by jemis-lakhani** [comment]
- Remove console logs.

**Pull request merged**